Source/CarthageKit/Project.swift

```diff
@@ -584,10 +584,13 @@ public final class Project {
 							}
 							return self.downloadMatchingBinaries(for: dependency, atRevision: revision, fromRepository: repository, client: Client(server: server, isAuthenticated: false))
 						}
-						.flatMap(.concat) { self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, commitish: revision, toolchain: toolchain) }
-						.on(completed: {
-							_ = try? FileManager.default.trashItem(at: checkoutDirectoryURL, resultingItemURL: nil)
-						})
+						.flatMap(.concat) {
+							return self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, commitish: revision, toolchain: toolchain)
+								.on(terminated: {
+									// We can receive the 'interrupted' event due to the 'take(first: 1)' below, so listen to 'terminated' to ensure we catch it
```

Comment:
That sounds reasonable. I was worried that it was sending an event for each framework it found, and didn't want to delete checkouts until it finished, but looking again, it just runs a `.then` with the unzipped directory after successful copying, so listening for a single value will work.